src/ucp/amo/amo.inl

```diff
@@ -161,6 +161,7 @@ ucp_amo_send_request(ucp_request_t *req, ucp_send_callback_t cb)
     ucs_trace_req("returning amo request %p, status %s", req,
                   ucs_status_string(status));
     req->send.cb = cb;
+    req->flags  |= UCP_REQUEST_FLAG_CALLBACK;
```

Comment:
minor: can create some macro/func for setting send callback, where flag will also be set. While it is just two lines, it looks like callback and flag should always be set together.